keras/callbacks.py

```diff
@@ -428,6 +429,48 @@ def on_epoch_end(self, epoch, logs=None):
                     self.model.save(filepath, overwrite=True)
 
 
+class DeadReluDetector(Callback):
+    def __init__(self, original_model, x_train):
+        if not isinstance(original_model, Sequential):
```

Comment:
maybe this can help you https://github.com/philipperemy/keras-visualize-activations/blob/master/read_activations.py
I used something similar to get internal activations for my own project (in non Sequential models).